### PR TITLE
Adds refresh token flow

### PR DESCRIPTION
**Pull request by andela-Kshittu**

### CODE REVIEW

**Comment by andela-Kshittu** [comment]
Note: We need to add `user_api_offline` scope to our prod tableau client before releasing this.

File: src/TableauConnector.js [hunk 1]
```diff
@@ -122,8 +123,8 @@ class TableauConnector {
     tableau.authType = tableau.authTypeEnum.custom
 
     this.authenticate()
-      .then(accessToken => this.validateAccessIfNeeded(accessToken))
-      .then(accessToken => {
+      .then(({accessToken, refreshToken}) => this.validateAccessIfNeeded(accessToken, refreshToken))
```
Comment (rflprr) [inline src/TableauConnector.js]: `accessToken` is being passed through this chain, but it isn't needed anywhere. I'd suggest passing `refreshToken` only.

File: src/auth.js [hunk 2]
```diff
@@ -43,25 +43,45 @@ const generateCodeVerifier = () => {
...
+    return refreshToken
   }
   return null
 }
 
-const storeApiKey = (key) => {
+const getAccessToken = async (useTableauPassword = false) => {
   if (window.localStorage) {
-    window.localStorage.setItem(apiTokenKey, key)
+    let refreshToken = window.localStorage.getItem(refreshTokenKey)
```
Comment (rflprr) [inline src/auth.js]: Reuse `getRefreshToken()` instead of duplicating the logic in it?

File: src/auth.js [hunk 3]
```diff
@@ -43,25 +43,45 @@ const generateCodeVerifier = () => {
...
+      refreshToken = window.tableau.password || refreshToken
+    }
+    if (refreshToken) { // exchange refresh token for access token
+      try {
+        const response = await api.getRefreshedTokens(refreshToken)
+        // store new refresh token
+        storeRefreshToken(response.data.refresh_token)
+        return response.data.access_token
+      } catch (error) {
+        return null
```
Comment (rflprr) [inline src/auth.js]: Logging this error would be important for troubleshooting.

File: src/auth.js (hunk 2 above)
Comment (andela-Kshittu) [reply src/auth.js]: Good catch!, thanks.

**Pull request merged**